File: spec/php/StrEncodingsTest.php
<?php
namespace Kaitai\Struct\Tests;

class StrEncodingsTest extends TestCase {
    public function testStrEncodings() {
        $r = StrEncodings::fromFile(self::SRC_DIR_PATH . "/str_encodings.bin");

        $this->assertEquals("Some ASCII", $r->str1());
        $this->assertEquals("こんにちは", $r->str2());
        $this->assertEquals("こんにちは", $r->str3());
        $this->assertEquals("░▒▓", $r->str4());
    }
}

File: spec/go/str_encodings_test.go
package spec

import (
	"os"
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"

	. "test_formats"
)

func TestStrEncodings(t *testing.T) {
	f, err := os.Open("../../src/str_encodings.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)

	var h StrEncodings
	err = h.Read(s, &h, &h)
	if err != nil {
		t.Fatal(err)
	}

	assert.Equal(t, "Some ASCII", h.Str1)
	assert.Equal(t, "こんにちは", h.Str2)
	assert.Equal(t, "こんにちは", h.Str3)
	assert.Equal(t, "░▒▓", h.Str4)
}
